src/main/java/com/matt/forgehax/mods/DerpMod.java

```diff
@@ -0,0 +1,244 @@
+          }))
+          .defaultTo(true)
+          .build();
+
+  public final Setting<Float> rotateChance =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("rotate_chance")
+          .description("Chance to rotate for every derp")
+          .min(0f)
+          .max(1f)
+          .defaultTo(1f)
+          .build();
+
+  public final Setting<Boolean> sneak =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("sneak")
+          .description("Toggle sneak every derp")
+          .changed(change -> MC.addScheduledTask(() -> {
+            if (!isEnabled()) return;
+            if (change.getFrom() && !change.getTo()) sneakDisabled();
+            if (!change.getFrom() && change.getTo()) {
+              EntityPlayer player = getLocalPlayer();
+              sneaking = player != null && player.isSneaking();
+            }
+          }))
+          .defaultTo(false)
+          .build();
+
+  public final Setting<Float> sneakChance =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("sneak_chance")
+          .description("Chance to sneak for every derp")
+          .min(0f)
+          .max(1f)
+          .defaultTo(1f)
+          .build();
+
+  public final Setting<Boolean> hit =
+      getCommandStub()
+          .builders()
+          .<Boolean>newSettingBuilder()
+          .name("hit")
+          .description("Randomly hit every derp")
+          .defaultTo(true)
+          .build();
+
+  public final Setting<Float> hitChance =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("hit_chance")
+          .description("Chance to hit for every derp")
+          .min(0f)
+          .max(1f)
+          .defaultTo(.25f)
+          .build();
+
+  public final Setting<Float> hitPan =
+      getCommandStub()
+          .builders()
+          .<Float>newSettingBuilder()
+          .name("hit_pan")
+          .description("Chance for any hit to be the off hand")
+          .min(0f)
+          .max(1f)
+          .defaultTo(.5f)
+          .build();
+
+  private void sendRotatePacket(EntityPlayer player, float pitch, float yaw) {
+    ignoreAndSend(new CPacketPlayer.Rotation(yaw, pitch, player.onGround));
+  }
+
+  private void rotateDisabled() {
+    EntityPlayer player = getLocalPlayer();
+    if (player == null) return;
+
+    sendRotatePacket(player, player.rotationPitch, player.rotationYawHead);
+  }
+
+  private CPacketEntityAction.Action actionFromSneaking(boolean sneaking) {
+    return sneaking
+        ? CPacketEntityAction.Action.START_SNEAKING
+        : CPacketEntityAction.Action.STOP_SNEAKING;
+  }
+
+  private void sendSneakPacket(EntityPlayer player, boolean sneaking) {
+    this.sneaking = sneaking;
+    ignoreAndSend(new CPacketEntityAction(player, actionFromSneaking(sneaking)));
```

Comment:
Just a heads up - no cheat and other anti-cheats will set the player back for moving too fast while sneaking. Also can you avoid using the ignoreAndSend method unless its necessary? There are mods in the background that keep track of a players state and doing this will prevent them from working properly.

Reply:
> no cheat and other anti-cheats will set the player back for moving too fast while sneaking

That's why sneak is toggleable. That's also why it defaults to false.

> There are mods in the background that keep track of a players state and doing this will prevent them from working properly.

That's the point. Derp is not supposed to change the player's state. You should be able to play normally with derp enabled, only other players will see the difference.